doc-tool/src/dotty/tools/dottydoc/staticsite/Site.scala

```diff
@@ -199,7 +200,8 @@ case class Site(
           if (e.kind == "package") ("/index.html", -1)
           else (".html", 0)
 
-        val target = mkdirs(fs.getPath(outDir.getAbsolutePath +  "/api/" + e.path.mkString("/") + suffix))
+        val filePath = e.path.map(NameTransformer.encodeSting).mkString("/")
```

Comment:
This should be fixed much earlier than this. `Entity#path` should always contain name-mangled strings. This involves changing at least [`factories#path`](https://github.com/lampepfl/dotty/blob/3eff8316fbf73c186beaf5f57089dca8acdb97a4/doc-tool/src/dotty/tools/dottydoc/model/factories.scala#L32-L35) to use `name.mangledString` instead of `name.show`.